include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -298,6 +298,52 @@ arena_dalloc_large(tsdn_t *tsdn, void *ptr, tcache_t *tcache, szind_t szind,
 	}
 }
 
+JEMALLOC_ALWAYS_INLINE size_t
+arena_slab_regind_impl(div_info_t* div_info, szind_t binind,
+    edata_t *slab, const void *ptr) {
+	size_t diff, regind;
+
+	/* Freeing a pointer outside the slab can cause assertion failure. */
+	assert((uintptr_t)ptr >= (uintptr_t)edata_addr_get(slab));
+	assert((uintptr_t)ptr < (uintptr_t)edata_past_get(slab));
+	/* Freeing an interior pointer can cause assertion failure. */
+	assert(((uintptr_t)ptr - (uintptr_t)edata_addr_get(slab)) %
+	    (uintptr_t)bin_infos[binind].reg_size == 0);
+
+	diff = (size_t)((uintptr_t)ptr - (uintptr_t)edata_addr_get(slab));
+
+	/* Avoid doing division with a variable divisor. */
+	regind = div_compute(div_info, diff);
+	return regind;
+}
+
+/*
+* Checks whether ptr is already deallocated in the arena.
+* Fires safety check if it has been deallocated and returns true.
+*/
+JEMALLOC_ALWAYS_INLINE bool
+arena_dalloc_small_sanity_check(tsdn_t *tsdn, void *ptr) {
```

Comment:
nit: naming -- let's stick to `_safety_check`.

Also in the comments and the PR title, let's avoid using the term "in the arena", since there are several things at the arena level. Probably say something like "using slab bitmap".

Reply:
Renamed